presto-common/src/main/java/com/facebook/presto/common/block/ArrayBlock.java

```diff
@@ -208,4 +208,24 @@ public Block getLoadedBlock()
                 offsets,
                 loadedValuesBlock);
     }
+
+    private boolean isSinglePositionBlock(int position)
+    {
+        // Verify the array has only one element
+        if (position == 0 && positionCount == 1) {
+            // verify offsets array has only one element.
+            return offsets.length == 2;
```

Comment:
`offsets.length == 2` can be wrapped into the `if` condition to return true here.